packages/core/utils/src/modules-sdk/medusa-service.ts

```diff
@@ -245,12 +245,7 @@ export function MedusaService<
         ): Promise<T | T[]> {
           const serviceData = Array.isArray(data) ? data : [data]
           const service = this.__container__[serviceRegistrationName]
-          const models = await service.update(serviceData, sharedContext)
-          const response = models.length
-            ? Array.isArray(data)
-              ? models
-              : models[0]
-            : []
+          const response = await service.update(serviceData, sharedContext)
```

Comment:
thought: I guess this is technically a breaking change? All service calls that have used the selector-based update will now return an array right?

Reply:
yes thats right, but it was actually a bug, so I consider it more as a fix than a breaking change. The types was describing that you would receive an array but in fact you were receiving an object. Also, this logic is handled in the internal service

Reply:
Yeah, that's true. Think it should be OK to treat it as a bug fix.